### PR TITLE
feat(hot reloading): Add first version of hot reloading

### PR DESCRIPTION
**Pull request by borismelnik**
Version for: https://github.com/status-im/status-desktop/issues/5690

I tried to make a partial updates, but looks like it possible just in special environment.
What do you think about it?

https://user-images.githubusercontent.com/82511785/168593471-a994d814-a0e8-4ef2-82a9-5696734061b0.mov

### Checklist

- [ ] follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
  - the scope should be the component's name e.g: `feat(StatusListItem): ... `
  - when adding new components, the scope is the module e.g: `feat(StatusQ.Controls): ...`
- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?
- [ ] is this a breaking change?
    - [ ] use the dedicated `BREAKING CHANGE` commit message section
    - [ ] resolve breaking changes in [status-desktop](https://github.com/status-im/status-desktop)
        - [ ] (pre-merge) adapt code to breaking changes
        - [ ] (post-merge) update StatusQ submodule pointer
- [ ] test changes in [status-desktop](https://github.com/status-im/status-desktop)

### CODE REVIEW

**Review by alexandraB99 — APPROVED** [review]
lgtm!

File: sandbox/sandboxapp.cpp [hunk 1]
```diff
@@ -21,6 +29,16 @@ void SandboxApp::startEngine()
 
 #ifdef QT_DEBUG
     const QUrl url = QUrl::fromLocalFile(applicationDirPath() + "/../main.qml");
+    m_watcher.addPath(applicationDirPath() + "/../");
+//    m_watcher.addPath(applicationDirPath() + "/../../src");
```
Comment (PascalPrecht) [inline sandbox/sandboxapp.cpp:33]: I guess this can be removed?

**Review by PascalPrecht — APPROVED** [review]
Approving, judging by the attached video.
However, this isn't really related to status-desktop, correct?
Asking cause it's linked in the commit..

Do we also want to bring back the "reload" button?

File: sandbox/sandboxapp.cpp [hunk 2]
```diff
@@ -12,6 +13,13 @@ SandboxApp::SandboxApp(int &argc, char **argv)
       m_handler(new Handler(this))
 {
     connect(m_handler, &Handler::restartQml, this, &SandboxApp::restartEngine, Qt::QueuedConnection);
```
Comment (saledjenic) [inline sandbox/sandboxapp.cpp:15]: @borismelnik what's the purpose of `m_handler`, I see the usage in the line 46, but except in this connection I don't see anywhere that the registered context property "app" is used nor `restartQml` signal sending?

**Review by stefandunca — APPROVED** [review]
Thanks, for bringing this back.

**Comment by endulab** [comment]
@borismelnik good first try. I found a nice article about more sophisticated approaches: https://qml.guide/live-reloading-hot-reloading-qml/. Partial updates are described here also.

**Review by endulab — APPROVED** [review]

File: sandbox/sandboxapp.cpp (hunk 2 above)
Comment (borismelnik) [reply sandbox/sandboxapp.cpp:15]: yeah its old code

File: sandbox/sandboxapp.cpp (hunk 1 above)
Comment (borismelnik) [reply sandbox/sandboxapp.cpp:33]: oh yeah sorry

**Review by MishkaRogachev — APPROVED** [review]
Perfect! Maybe in the future we will need two modes: developer with hot reload and production with compiling qml's into binary.

**Comment by PascalPrecht** [comment]
@borismelnik What is the status of this PR?

**Review by saledjenic — APPROVED** [review]

**Pull request merged**